packages/react/src/components/Selector/index.js

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+import UiText from '../UiText';
+
+const Selector = React.forwardRef(
+  (
+    {
+      className,
+      disabled: isDisabled,
+      id,
+      onChange,
+      options,
+      defaultIndex,
+      ...props
+    },
+    ref,
+  ) => {
+    const [activeIndex, setActiveIndex] = useState(defaultIndex);
+
+    const handleOptionClick = (index, option) => {
+      if (isDisabled || option.disabled) return;
+      if (index === activeIndex) {
+        setActiveIndex(null);
+        onChange(null);
+        return;
+      }
+
+      setActiveIndex(index);
+      onChange(option.value);
+    };
+
+    return (
+      <div className='f-FormEl-wrapper'>
+        <UiText
+          as='div'
+          variant={UiText.VARIANTS.content}
+          className={cx('f-Selector', className, {
+            'is-disabled': isDisabled,
+          })}
+          disabled={isDisabled}
+          id={id}
+          name={id}
+          onChange={onChange}
+          ref={ref}
+          {...props}
+        >
+          {options.map((option, index) => (
+            <div
+              key={option.value}
+              role='button'
+              onClick={() => handleOptionClick(index, option)}
+              onKeyPress={({ which }) =>
+                which === 13 ? handleOptionClick(index, option) : undefined
+              }
+              className={cx('f-Selector-item', {
+                'is-active': activeIndex === index,
+                'is-disabled': option.disabled === true,
+              })}
+              tabIndex={0}
+            >
+              {option.label}
+            </div>
+          ))}
+        </UiText>
+      </div>
+    );
+  },
+);
+
+Selector.displayName = 'Selector';
+
+Selector.propTypes = {
+  className: PropTypes.string,
+  disabled: PropTypes.bool,
+  id: PropTypes.string.isRequired,
+  onChange: PropTypes.func.isRequired,
+  defaultIndex: PropTypes.number,
+  options: PropTypes.arrayOf(
+    PropTypes.exact({
+      disabled: PropTypes.bool,
+      label: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+        .isRequired,
+      value: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
+        .isRequired,
+    }),
+  ),
+};
+
+Selector.defaultProps = {
+  className: undefined,
+  disabled: false,
+  defaultIndex: null,
```

Comment:
```suggestion
  defaultIndex: 0,
```

Do we want to have the first item selected by default?

Reply:
Nope, apparently not, it's optional